### ISSUE TITLE
AWS: Unable to SSH into cluster created with shared private subnet

### ISSUE THREAD

**Issue by gopinatht** [kubernetes/kops#1476]
We cannot access the nodes or the master created in a shared VPC with pre-existing private subnet because the security groups for the master and nodes do not support port 22. Here is what my config looks like

```
apiVersion: kops/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: "2017-01-14T00:10:01Z"
  name: <URL>
spec:
  api:
    loadBalancer:
      type: Internal
  channel: alpha
  cloudProvider: aws
  configBase: s3://S3STORE/S3NAME
  etcdClusters:
  - etcdMembers:
    - instanceGroup: master-us-east-1e
      name: us-east-1e
    name: main
  - etcdMembers:
    - instanceGroup: master-us-east-1e
      name: us-east-1e
    name: events
  kubernetesApiAccess:
  - 0.0.0.0/0
  kubernetesVersion: v1.5.1
  masterInternalName: api.internal.<URL>
  masterPublicName: api.<URL>
  networkCIDR: 10.44.76.0/23
  networkID: vpc-xxxxxxxx
  networking:
    weave: {}
  nonMasqueradeCIDR: 100.64.0.0/10
  sshAccess:
  - 0.0.0.0/0
  subnets:
  - cidr: 10.44.76.0/25
    id: subnet-xxxxxxxx
    name: us-east-1e
    type: Private
    zone: us-east-1e
  - cidr: 10.44.76.0/25
    id: subnet-xxxxxxxx
    name: utility-us-east-1e
    type: Utility
    zone: us-east-1e
  topology:
    dns:
      type: Public
    masters: private
    nodes: private
```

I spoke to @justinsb on slack and he suggested that I open an issue so here it is.

**Comment by chrislovecnm** [comment]
Was a bastion created?

**Comment by justinsb** [comment]
So currently with `--topology private` we prevent SSH access to the masters and nodes, other than through a bastion instance.  It does seem that this is actually a request for another mode, where we allow SSH access to the masters and nodes to anyone that is on the VPC (e.g. has VPN-ed in to it).  Is that right @gopinatht ?

We will have a workaround, which is that you will be able to attach another security group once #1444 merges, which will allow any additional access you want, I believe.

**Comment by gopinatht** [comment]
@justinsb Yes, especially in a private network, it would help to be able to directly SSH into master and nodes and not need a jump server. The intent is to allow access only from the org network so a public facing bastion is not necessary or recommended.

**Comment by jkinkead** [comment]
This is how we've set things up, too, and my current workaround requires using an existing subnet or manually editing the route table after the subnet comes up. It's not ideal.

**Comment by geojaz** [comment]
closed in #1444
